### PR TITLE
[1.x] Added js directory path alias

### PR DESCRIPTION
**Pull request by slavarazum**
With `@` alias for js folder, we can use absoulte path instead of relative.

Imports will be more clear:

```diff
-import JetLabel from './../../Jetstream/Label'
+import JetLabel from '@/Jetstream/Label'
```

Its standard practice from vue cli preset. Also, you can check out article [https://laravel-news.com/laravel-mix-alias](https://laravel-news.com/laravel-mix-alias) with example of implementation.

### CODE REVIEW

File: stubs/webpack.mix.js [hunk 1]
```diff
@@ -15,4 +16,11 @@ mix.js('resources/js/app.js', 'public/js')
     .postCss('resources/css/app.css', 'public/css', [
         require('postcss-import'),
         require('tailwindcss'),
-    ]);
+    ])
+    .webpackConfig({
+        resolve: {
+            alias: {
+                '@': path.resolve(__dirname, 'resources/js/')
```
Comment (reinink) [inline stubs/webpack.mix.js:23]: I wonder if this should be shortened to:

```js
'@': path.resolve('resources/js'),
```

(That's what I've [always done](https://github.com/inertiajs/pingcrm/blob/master/webpack.mix.js#L40).)

**Comment by claudiodekker** [comment]
I personally do really like this, but I do have two concerns:
- Aliasing to `@` does cause issues with discovery in PHPStorm (and perhaps other editors?) [There are workarounds for this](https://stefanbauer.me/tips-and-tricks/autocompletion-for-webpack-path-aliases-in-phpstorm-when-using-laravel-mix), but this makes me doubt whether this is a bit _too opinionated_ to be a default?
- This file is shared with Livewire. Are we sure it doesn't cause issues there either?

Regarding the file being shared, [I already mentioned this earlier today on this issue](https://github.com/laravel/jetstream/issues/326#issuecomment-705566445), so splitting it might be the way forward anyway.

**Comment by claudiodekker** [comment]
Additionally, it might make more sense to target v2 (`master`), as this is not really a bugfix, and even more importantly because there's newly added Inertia views there (see #298) that we'd otherwise need to create a separate PR for anyway.

File: stubs/webpack.mix.js (hunk 1 above)
Comment (slavarazum) [reply stubs/webpack.mix.js]: Just tested, it works 👍
Pushed the update.

**Comment by slavarazum** [comment]
@claudiodekker This update haven't any breaking changes and doesn't force you to use alias.

P.S. I have successfully setup alias resolving in VSCode and PHPStrom

**Comment by taylorotwell** [comment]
Maybe we should break out the webpack stubs to be different for Inertia vs. Livewire? Also, any more thoughts on the PHP Storm issue? I don't use PHP Storm so can't really comment on that.

**Comment by slavarazum** [comment]
Root directory alias might be useful for both presets. PHPStorm supports webpack.config detection with alias resolving.

**Comment by claudiodekker** [comment]
@slavarazum Yes, that applies to `webpack.config.js`, but not to the `webpack.mix.js` that Laravel (Mix) uses & is being changed here.

One way around it [that I linked earlier](https://stefanbauer.me/tips-and-tricks/autocompletion-for-webpack-path-aliases-in-phpstorm-when-using-laravel-mix) is to create a `webpack.config.js` (that PHPStorm can use) and then to import that config into the `webpack.mix.js`, but I don't see that being implemented here, hence this being an issue.

**Comment by taylorotwell** [comment]
@claudiodekker @slavarazum where do we go from here? We probably need to solve this PHP Storm issue that @claudiodekker is raising.

**Comment by slavarazum** [comment]
@taylorotwell Let me export webpack config to separate file as @claudiodekker described.

**Pull request merged**